components/PageControl/src/MDCPageControl.m

```diff
@@ -193,6 +194,7 @@ - (void)setCurrentPage:(NSInteger)currentPage
     [CATransaction setDisableActions:YES];
     [_indicators[previousPage] setHidden:NO];
     [CATransaction commit];
+    [self sendActionsForControlEvents:UIControlEventValueChanged];
```

Comment:
I believe the CATransaction is asynchronously dispatched into the main run loop. I think we should place the `sendActionsForControlEvents:` into the completion block of the transaction to ensure that it's only invoked once the changes are actually made.

```objectivec
    dispatch_async(dispatch_get_main_queue(), ^{
      NSLog(@"Dispatched before");
    });
    [CATransaction begin];
    [CATransaction setDisableActions:YES];
    [_indicators[previousPage] setHidden:NO];
    [CATransaction setCompletionBlock:^{
      NSLog(@"Completion block");
    }];
    [CATransaction commit];
    dispatch_async(dispatch_get_main_queue(), ^{
      NSLog(@"Dispatched after");
    });

    NSLog(@"Subsequent code");
```

```
2017-08-07 16:29:03.806 MDCCatalog[61661:7937194] Subsequent code
2017-08-07 16:29:03.807 MDCCatalog[61661:7937194] Dispatched before
2017-08-07 16:29:03.807 MDCCatalog[61661:7937194] Completion block
2017-08-07 16:29:03.807 MDCCatalog[61661:7937194] Dispatched after
```